omegatrade-app/backend/app/controller/user.controller.js

```diff
@@ -11,59 +12,30 @@ const { v4: uuidv4 } = require('uuid');
  * @param {JSON} req contains request headers and payload businessEmail,fullName,password,photoUrl,provider
  */
 exports.register = async function (req, res) {
-    const body = req.body;
-    await User.findUser(body.businessEmail, function (err, data) {
-        if (data && data.userId) {
-            res.status(400).json({
-                success: false,
-                message: 'Email Already Exists',
-            });
-        }
-        if (err) {
-            res.status(400).json({
-                success: false,
-                message: 'Something went wrong while fetching user!'
-            });
-        }
-    });
-    const salt = await bcrypt.genSalt(10);
-    const password = await bcrypt.hash(body.password, salt);
-    delete body.password; 
-    delete body.confirmPassword;
-    const user = {
-        ...body,
-        userId: uuidv4(),
-    }
     try {
-        await User.registerUser({ ...user,password }, function (err, data) {
+        const body = req.body;
+        const [registerUser] = await User.findUser(body.businessEmail);
+        if (registerUser && registerUser.userId) {
+            return res.status(400).json({ success: false, message: 'Registration failed, Email already exists!' });
```

Comment:
Use `409` to be consistent with other similar functions (or change all the other to use `400`, but I would prefer `409` for these cases, as the error message also indicates a conflict.)

Reply:
the above issue has been resolved.

Reply:
It's not really clear to me how this has been resolved. The error code here is still 400, while 409 is also used in a number of places. While it's not really a big issue, it seems somewhat randomly chosen in a number of places in these files. So to be sure we're on the same page:
- Code `400` indicates Bad Request and is a relatively general error code that indicates that the client sent a request with invalid data.
- Code `409` indicates Conflict, and is a more specific error code that indicates that for example a user with the same email address already exists.

I would say:
- Use `409` when a record with the same id/code/... exists (and it is not something that we should not tell the user about).
- Use `400` when a more general error was detected in the input, or when we do not want to inform the user about the specific error, for example if it would reveal security-related information.